src/Microsoft.AspNet.Mvc.Razor/Compilation/CompilationResult.cs

```diff
@@ -13,119 +11,49 @@ namespace Microsoft.AspNet.Mvc.Razor
     /// </summary>
     public class CompilationResult
     {
-        private Type _type;
-
         /// <summary>
-        /// Creates a new instance of <see cref="CompilationResult"/>.
+        /// Creates a new instance of <see cref="CompilationResult"/> that represents a success in compilation.
         /// </summary>
-        protected CompilationResult()
+        /// <param name="type">The compiled type.</param>
+        public CompilationResult([NotNull] Type compiledType)
         {
+            CompiledType = compiledType;
         }
 
         /// <summary>
-        /// Gets the path of the Razor file that was compiled.
+        /// Creates a <see cref="CompilationResult"/> that represents a failure in compilation.
         /// </summary>
-        public string FilePath
+        /// <param name="compilationFailure">The <see cref="ICompilationFailure"/> produced from parsing or
+        /// compiling the Razor file.</param>
+        public CompilationResult([NotNull] ICompilationFailure compilationFailure)
         {
-            get
-            {
-                if (File != null)
-                {
-                    return File.PhysicalPath;
-                }
-                return null;
-            }
+            CompilationFailure = compilationFailure;
         }
 
         /// <summary>
-        /// Gets a sequence of <see cref="CompilationMessage"/> instances encountered during compilation.
+        /// Gets the type produced as a result of compilation.
         /// </summary>
-        public IEnumerable<CompilationMessage> Messages { get; private set; }
+        /// <remarks>This property is null when compilation fails.</remarks>
+        public Type CompiledType { get; }
 
         /// <summary>
-        /// Gets (or sets in derived types) the generated C# content that was compiled.
+        /// Gets the type produced as a result of compilation.
```

Comment:
this comment matches the summary above.  suspect you mean `Gets the <see cref="ICompilationFailure"/> produced from parsing or compiling the Razor file.`